components/gem/actions/create-candidate/create-candidate.mjs

```diff
@@ -0,0 +1,171 @@
+      optional: true,
+    },
+    workInfo: {
+      type: "string",
+      label: "Work Info",
+      description: "A list of objects containing candidate's work information. **Format: [{\"company\": \"string\", \"title\": \"string\", \"work_start_date\": \"string\", \"work_end_date\": \"string\", \"is_current\": \"string\"}]**. [See the documentation](https://api.gem.com/v0/reference#tag/Candidates/paths/~1v0~1candidates/post) for further details.",
+      optional: true,
+    },
+    profileUrls: {
+      type: "string[]",
+      label: "Profile URLs",
+      description: "If `Profile URLs` is provided with an array of urls, social `profiles` will be generated based on the provided urls and attached to the candidate",
+      optional: true,
+    },
+    phoneNumber: {
+      type: "string",
+      label: "Phone Number",
+      description: "Candidate's phone number",
+      optional: true,
+    },
+    projectIds: {
+      propDefinition: [
+        gem,
+        "projectIds",
+      ],
+      optional: true,
+    },
+    customFields: {
+      type: "string[]",
+      label: "Custom Fields",
+      description: "Array of objects containing new custom field values. Only custom fields specified in the array are updated. **Format: [{\"custom_field_id\": \"string\", \"value\": \"string\"}]**. [See the documentation](https://api.gem.com/v0/reference#tag/Candidates/paths/~1v0~1candidates/post) for further details.",
+      optional: true,
+    },
+    sourcedFrom: {
+      type: "string",
+      label: "Sourced From",
+      description: "Where the candidate was sourced from",
+      options: SOURCED_FROM,
+      optional: true,
+    },
+    autofill: {
+      type: "boolean",
+      label: "Autofill",
+      description: "Requires `Linked In Handle` to be non-null. Attempts to fill in any missing fields.",
+      optional: true,
+    },
+  },
+  async run({ $ }) {
+    const emails = [
+      {
+        email_address: this.primaryEmail,
+        is_primary: true,
+      },
+    ];
+    if (this.additionalEmails) emails.push(...this.additionalEmails.map((email) => ({
+      email_address: email,
+      is_primary: false,
+    })));
+
+    if (emails.length === 0) {
+      throw new Error("Primary Email Address is required");
+    }
+    const candidate = await this.gem.createCandidate({
+      $,
+      data: {
+        created_by: this.createdBy,
+        first_name: this.firstName,
+        last_name: this.lastName,
+        nickname: this.nickname,
+        emails,
+        linked_in_handle: this.linkedInHandle,
+        title: this.title,
+        company: this.company,
+        location: this.location,
+        school: this.school,
+        educational_info: this.educationalInfo,
+        work_info: this.workInfo,
+        profile_urls: this.profileUrls,
+        phone_number: this.phoneNumber,
+        project_ids: this.projectIds,
+        custom_fields: this.customFields,
```

Comment:
Should create a utils function to parse the JSON in `eductionalInfo`, `workInfo`, and `customFields`.
```suggestion
        educational_info: utils.parseJson(this.educationalInfo),
        work_info: utils.parseJson(this.workInfo),
        profile_urls: this.profileUrls,
        phone_number: this.phoneNumber,
        project_ids: this.projectIds,
        custom_fields: utils.parseJson(this.customFields),
```